pageserver/src/walredo.rs

```diff
@@ -653,7 +667,7 @@ impl PostgresRedoProcess {
             // If we have more data to write, wake up if 'stdin' becomes writeable or
             // we have data to read. Otherwise only wake up if there's data to read.
             let nfds = if nwrite < writebuf.len() { 3 } else { 2 };
-            let n = nix::poll::poll(&mut pollfds[0..nfds], TIMEOUT.as_millis() as i32)?;
+            let n = nix::poll::poll(&mut pollfds[0..nfds], wal_redo_timeout.as_millis() as i32)?;
```

Comment:
Just a note (it's not the issue of the current PR, I just wanted to highlight the behaviour): `as_millis` returns `u128` and casting it to `i32` will _silently_ truncate the value.

```rust
let i32_max = i32::MAX as u64;
dbg!(Duration::from_millis(i32_max).as_millis() as i32);
dbg!(Duration::from_millis(i32_max + 1).as_millis() as i32);
```

outputs 
```
[src/main.rs:6] Duration::from_millis(i32_max).as_millis() as i32 = 2147483647
[src/main.rs:7] Duration::from_millis(i32_max + 1).as_millis() as i32 = -2147483648
```

Not sure that it's what we want :)
We could use https://doc.rust-lang.org/std/convert/trait.TryFrom.html to explicitly produce such conversion errors.

On the other hand, `i32::MAX` is roughly 596 hours, so not sure that's so critical to fix in this PR, feels like no sane person would want to use such timeouts.